doc/source/security/using-keyring-plugin.rst

```diff
@@ -0,0 +1,124 @@
+
+We recommend the plugin should be loaded in the configuration file to facilitate
+recovery for encrypted tables. Also, the redo log and undo log encryption cannot
+be used without ``--early-plugin-load``. The normal plugin load happens too late
+in startup. 
+
+To use the the keyring_vault, You can add this option to your
+configuration file:
+
+.. code-block:: guess
+
+    [mysqld]
+    early-plugin-load="keyring_vault=keyring_vault.so"
+    loose-keyring_vault_config="/home/mysql/keyring_vault.conf"
+
+.. note::
+
+    The keyring_vault extension, ".so" and the file location for the vault configuration should be changed to match your operating system's extension and operating system location. 
+
+You could also run the following command which loads the keyring_file plugin:
+
+.. code-block:: bash
+
+   $ mysqld --early-plugin-load="keyring_file=keyring_file.so"
+
+.. warning::
+
+   Only one keyring plugin should be enabled at a time. Enabling multiple keyring plugins is not supported and may result in data loss.
+
+.. note::
+
+     If server is started with different plugins loaded early, the ``--early-plugin-load`` option should contain the plugin names in a double-quoted list with each plugin name separated by a semicolon. The use of double quotes ensures the semicolons do not create issues when the list is executed in a script.
+
+.. seealso::
+
+   |MySQL| Documentation:
+      - `Installing a Keyring Plugin <https://dev.mysql.com/doc/refman/8.0/en/keyring-installation.html>`_
+      - `The --early-plugin-load Option <https://dev.mysql.com/doc/refman/8.0/en/server-options.html#option_mysqld_early-plugin-load>`_
+
+Apart from installing the plugin you also must set the :variable:`keyring_vault_config` variable to point to the keyring_vault configuration file.
+
+The `keyring_vault_config` file has the following information:
+
+* ``vault_url`` - the Vault server address
+
+* ``secret_mount_point`` - the mount point name where the `keyring_vault` stores the keys
+
+* ``token`` - a token generated by the Vault server
+
+* ``vault_ca [optional]`` - if the machine does not trust the Vault's CA certificate, this variable points to the CA certificate used to sign the Vault's certificates
+
+An example of the configuration file looks like this: ::
+
+  vault_url = https://vault.public.com:8202
+  secret_mount_point = secret
+  token = 58a20c08-8001-fd5f-5192-7498a48eaf20
+  vault_ca = /data/keyring_vault_confs/vault_ca.crt
+
+.. warning::
+
+    Each ``secret_mount_point`` must be used by only one server. If multiple server use the same secret_mount_point, the behavior is unpredictable.
+
+The first time a key is fetched from a `keyring`, the `keyring_vault` communicates with the Vault server to retrieve the key type and data.
+
+A key deletion is only possible with the use of the keyring_udf plugin and deletes the key from the in-memory hash map and the Vault server.
```

Comment:
A key deletion (of user created keys - it is not possible to delete system keys like Master Key).

Reply:
Fixed.